katdal/test/test_chunkstore_s3.py

```diff
@@ -411,12 +418,25 @@ def do_all(self):
             if key.lower() not in HOP_HEADERS.union({'date', 'server'}):
                 self.send_header(key, value)
         self.end_headers()
-        self.wfile.write(content[:-10] if truncate else content)
+        if read_pause:
+            halfway = len(content) // 2
+            self.wfile.write(content[:halfway])
+            time.sleep(read_pause)
```

Comment:
Do you need to flush the file before sleeping, or is it unbuffered?

Reply:
The `socketserver.StreamRequestHandler` base class of `http.server.BaseHTTPRequestHandler` defines `wfile` as an unbuffered `socketserver._SocketWriter` if the `wbufsize` class attribute is set to 0 (and it is).

I'm adding a flush anyway since it cannot hurt and it only gets called in a corner case where it matters most.